### DISCUSSION TITLE
How to change case of multiple directories recursively

### DISCUSSION THREAD

**Discussion by blueray453** [sharkdp/fd#1338] [Q&A] [answered]
I am using [ccase](https://github.com/rutrum/convert-case). The following command works for renaming a single directory.

    mv camelCase (ccase -t Kebab camelCase)

If I want to recursively rename all directories to kebab case then what would be the command.

I tried:

    fd --type d -X  mv {} (ccase -t Kebab {})

and 

    fd --type d -X  mv {} $(ccase -t Kebab {})

giving error: `[fd error]: Only one placeholder allowed for batch commands`

**Comment by tmccombs** [comment, marked as answer]
I don't think you can do a bulk mv like that. You'll probably need to use -x instead. And you'll need to invoke a shell in order to call a nested command.


Something like:

```
fd --type d -x  bash -c 'mv {} $(ccase -t Kebab {})'
```

**Reply by blueray453** [reply]
Here is the directories:

    mkdir -p {PascalCase{1..3}{1..3},camelCase}/PascalCase/{PascalCase,camelCase}{1..5} && touch tmpdir/{PascalCase{1..3}{1..3},camelCase}/PascalCase/{PascalCase,camelCase}{1..5}/test.txt

your command work for `--max-depth 1`.

    fd --type d --max-depth 1 -x bash -c 'mv {} $(ccase -t Kebab {})'

When depth is more that 1 it give error:

    mv: cannot stat './PascalCase31/PascalCase/PascalCase1': No such file or directory

Another issue:

If I run `fd --type d --max-depth 1 -x bash -c 'mv {} $(ccase -t Kebab {})'` twice then it says:

    mv: cannot move './pascal-case-21' to a subdirectory of itself, './pascal-case-21/pascal-case-21'

Is there anything we can do about this. I mean something like, if file with this name does not exist then rename file.

**Comment by blueray453** [comment]
I think something like `-depth` and `-execdir` of find command will solve this problem.

> -depth Process each directory's contents before the directory itself.  

and

 >  -execdir command {} +
 >         Like -exec, but the specified command is run from the
 >         subdirectory containing the matched file, which is not
 >         normally the directory in which you started find. 

Is there anything similar

**Reply by blueray453** [reply]
The following command solved my problem:

    find . -depth -type d -execdir bash -c 'mv {} $(ccase -t Kebab {})' \;

Thank you very much for guiding me towards `bash -c`, however, not sure why i can use `{}` twice when inside `bash -c`
